go/l1client/ethereum_client.go

```diff
@@ -0,0 +1,209 @@
+package l1client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/obscuronet/obscuro-playground/go/l1client/rollupcontractlib"
+	"github.com/obscuronet/obscuro-playground/go/l1client/wallet"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var connectionTimeout = 15 * time.Second
+
+type ethNode struct {
```

Comment:
@otherview I feel that this class should be called `ethereumClientImpl` or similar, because that's what it is. Then to avoid confusion, the `client` field should be called `gethClient`. And I think we can push out the `chainID`, `wallet` fields etc., because AFAICT they're only used when deploying the rollup contract.

(But I realise Tudor is potentially telling you the opposite?)

Reply:
Good point - can you guys review the latest code ?

Reply:
Resolved in chat 👍